kratos/python_scripts/restart_utility.py

```diff
@@ -146,6 +150,14 @@ def IsRestartOutputStep(self):
         else:
             return (self.model_part.ProcessInfo[KratosMultiphysics.STEP] >= self.next_output)
 
+    def CreateOutputFolder(self):
+        if self.save_restart_files_in_folder:
+            folder_path = self.__GetFolderPathSave()
+            if not os.path.isdir(folder_path) and self.model_part.GetCommunicator().MyPID() == 0:
+                os.makedirs(folder_path)
+            self.model_part.GetCommunicator().Barrier()
```

Comment:
As a comment, some functions here rely on the model_part communicator for MPI, others (like the init) use the "global" DataCommunicator. I am aware that it is probably not possible to do this 100% of the time with the current state of our MPI but, in order to be able to support custom/split MPI communicators I believe we need to be more consistent (a given class should always rely on a single communicator).

Reply:
I am not clear on this one, maybe we can discuss in person

So you mean that we should use the data-comm always?

Reply:
Honestly I do not know what is the best approach yet, but the key point is that in most cases you would like to use a single communicator throughout the class, either the default one, one given by the user or the one defined by the ModelPart. Otherwise you might end up with strange parallelism errors where different parts of a given function work with different communicators.